src/ui/pages/Scan/Scan.tsx

```diff
@@ -93,6 +95,7 @@ const Scan = () => {
         setIsValueCaptured={setIsValueCaptured}
         handleReset={handleAfterScan}
         cameraDirection={cameraDirection}
+        onCheckPermissionFinish={setEnableCameraDirection}
```

Comment:
Shouldn't we only be enabling this if the permission has been accepted, and not just when we've finished checking the permissions?

Reply:
also looking at your video, it seems enabled on full page scanner but disabled on this scan tab, not sure why

Reply:
No it is disable all page. It's css issue

Reply:
`Shouldn't we only be enabling this if the permission has been accepted, and not just when we've finished checking the permissions?`

`onCheckPermissionFinish` return camera permission result. And I set it to `enableCameraDirection` state to disable/enable camera direction button.

Therefore, If camera permission is true => camera direction button will be enable.
Therefore, If camera permission is false => camera direction button will be disable.

Reply:
ah yes, of course, its directly passed to the function - my bad. thanks!